packages/ksf-pause-subscription/src/KSF/PauseSubscription/Component.purs

```diff
@@ -0,0 +1,233 @@
+import Data.Maybe (Maybe(..), isNothing)
+import Data.Nullable (Nullable, toMaybe, toNullable)
+import Data.Time.Duration as Time.Duration
+import Effect (Effect)
+import Effect.Aff as Aff
+import Effect.Class (liftEffect)
+import Effect.Class as Aff
+import Effect.Class.Console as Console
+import Effect.Now as Now
+import Effect.Uncurried (EffectFn1, mkEffectFn1)
+import KSF.Grid as Grid
+import KSF.Login.Component as Login
+import Persona as Persona
+import React.Basic (JSX, ReactComponent, StateUpdate(..), element, make, runUpdate)
+import React.Basic as React
+import React.Basic.DOM as DOM
+import React.Basic.DOM.Events (preventDefault)
+import React.Basic.Events (handler, handler_)
+import React.Basic.Extended (Style)
+import React.Basic.Extended as React.Extended
+import Unsafe.Coerce (unsafeCoerce)
+
+foreign import pauseSubscriptionStyles :: Style
+foreign import datePicker_ :: Fn0 (ReactComponent DatePickerProps)
+
+-- TODO: Move this into a separate package/component
+type DatePickerProps =
+  { onChange  :: EffectFn1 (Nullable JSDate) Unit
+  , className :: String
+  , value     :: Nullable JSDate
+  , format    :: String
+  , required  :: Boolean
+  , minDate   :: Nullable JSDate
+  , maxDate   :: Nullable JSDate
+  , disabled  :: Boolean
+  , locale    :: String
+  }
+
+type Self = React.Self Props State
+
+type Props =
+  { subsno    :: Int
+  , userUuid  :: Persona.UUID
+  , onCancel  :: Effect Unit
+  , onLoading :: Effect Unit
+  , onSuccess :: Persona.Subscription -> Effect Unit
+  , onError   :: Effect Unit
+  }
+
+type State =
+  { startDate    :: Maybe DateTime
+  , minStartDate :: Maybe DateTime
+  , endDate      :: Maybe DateTime
+  , minEndDate   :: Maybe DateTime
+  , maxEndDate   :: Maybe DateTime
+  }
+
+data Action
+  = SetStartDate (Maybe DateTime)
+  | SetMinStartDate (Maybe DateTime)
+  | SetEndDate (Maybe DateTime)
+
+pauseSubscription :: Props -> JSX
+pauseSubscription = make component { initialState, render, didMount }
+
+component :: React.Component Props
+component = React.createComponent "PauseSubscription"
+
+initialState :: State
+initialState =
+  { startDate: Nothing
+  , minStartDate: Nothing
+  , endDate: Nothing
+  , minEndDate: Nothing
+  , maxEndDate: Nothing
+  }
+
+update :: Self -> Action -> StateUpdate Props State
+update self action = Update $ case action of
+  SetStartDate newStartDate ->
+    self.state
+      { startDate = newStartDate
+      , minEndDate = calcMinEndDate newStartDate
+      , maxEndDate = calcMaxEndDate newStartDate
+      }
+  SetEndDate newEndDate -> self.state { endDate = newEndDate }
+  SetMinStartDate newMinStartDate -> self.state { minStartDate = newMinStartDate }
+
+-- | Minimum pause period is one week
+calcMinEndDate :: Maybe DateTime -> Maybe DateTime
+calcMinEndDate Nothing = Nothing
+calcMinEndDate (Just startDate) = do
+  let week = Time.Duration.Days 7.0
+  adjust week startDate
+
+-- | Maximum pause period is three months
```

Comment:
Do we have business rules for how much time there has to be between two pauses?

Reply:
Good question. There should be at least one week in between pauses (cc @kirstipa). Although this is not easy to show for the user, as the _amazing_ date picker does not support disabling specific dates. A proper error message goes a long way though.